cmd/crc/cmd/status.go

```diff
@@ -0,0 +1,136 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/docker/go-units"
+	"github.com/spf13/cobra"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/errors"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/oc"
+	"github.com/code-ready/machine/libmachine"
+	"github.com/code-ready/machine/libmachine/drivers"
+	"github.com/code-ready/machine/libmachine/state"
+)
+
+func init() {
+	rootCmd.AddCommand(statusCmd)
+}
+
+var statusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Display status of the cluster",
+	Long:  "Show details about OpenShift cluster and crc vm",
+	Run: func(cmd *cobra.Command, args []string) {
+		runStatus()
+	},
+}
+
+var statusFormat = `CRC VM:     {{.CrcStatus}}
+OpenShift:  {{.OpenShiftStatus}}
+DiskUsage:  {{.DiskUsage}}
+CacheUsage: {{.CacheUsage}} (used by oc binary, Disk image)
+`
+
+type Status struct {
+	CrcStatus       string
+	OpenShiftStatus string
+	DiskUsage       string
+	CacheUsage      string
+}
+
+func runStatus() {
+	// get vm status
+	api := libmachine.NewClient(constants.MachineBaseDir, constants.MachineCertsDir)
+	_, err := api.Exists(constants.DefaultName)
+	if err != nil {
+		logging.ErrorF("Failed while trying to check if the vm: %s exists", constants.DefaultName)
+	}
+	/*if exists {
+		host, err := api.Load(constants.DefaultName)
+		if err != nil {
+			errors.ExitWithMessage(1, err.Error())
+		}
+		state, err := host.Driver.GetState()
+		if err != nil {
+			errors.ExitWithMessage(1, fmt.Sprintf("Error fetching cluster status: %s", err.Error()))
+		}
+	}*/
+
+	openshiftStatus := "Stopped"
```

Comment:
I do not need to look further. I already see there is too much logic in wrong level,

Reply:
Right, i need to push the new changes, i was waiting for #237 to get in, since i am using the `oc.ClusterOperatorsAvailable` function from that PR.

Reply:
unfortunately this one was postponed due to an issue with the operator state that the cluster returns after the API server is just up; this state is unreliable.

We would look into the situation and try to clean this state... if not, something else needs to be done :-s

Reply:
I have requested to split out the `oc.ClusterOperatorStatus` related changes in a separate commit in that PR (https://github.com/code-ready/crc/pull/242#issuecomment-509517560), the wait period related discussion is independent of those changes, as that is only getting the status of the cluster operators and returning that.